src/core/ext/filters/client_channel/service_config.h

```diff
@@ -120,9 +149,16 @@ class ServiceConfig : public RefCounted<ServiceConfig> {
       grpc_json* json, CreateValue<T> create_value,
       typename SliceHashTable<RefCountedPtr<T>>::Entry* entries, size_t* idx);
 
+  static constexpr int kMaxParsers = 32;
+  static int registered_parsers_count;
+  static ServiceConfigParser registered_parsers[kMaxParsers];
```

Comment:
We'll need a way to initialize and clear this list when `grpc_init()` and `grpc_shutdown()` run.  I suggest using the same pattern that we use to register resolvers and LB policies.

Reply:
done